google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -1463,6 +1677,19 @@ public ByteArray getBytes() {
       return value.getByteArray();
     }
 
+    @Override
+    public <T extends AbstractMessage> T getProtoMessage(T m) {
+      Preconditions.checkNotNull(
+          m,
+          "Proto message may not be null. Use MyProtoClass.getDefaultInstance() as a parameter value.");
+      checkNotNull();
+      try {
+        return (T) m.toBuilder().mergeFrom(value.getByteArray().toByteArray()).build();
```

Comment:
Here also: Any possibility that we can do this conversion without copying to a byte array multiple times?

Reply:
@olavloite I have applied the optimization suggested above to this case too. Please let me know if this is correct.
```
value.getByteArray().toByteArray()
```
is changed to 
```
Base64.getDecoder()
                .wrap(
                    CharSource.wrap(value.getBase64String())
                        .asByteSource(StandardCharsets.UTF_8)
                        .openStream())
```